ui/components/app/snaps/snap-ui-address-input/snap-ui-address-input.tsx

```diff
@@ -0,0 +1,202 @@
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  FontWeight,
+  IconColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import Jazzicon from '../../../ui/jazzicon';
+
+export type SnapUIAddressInputProps = {
+  name: string;
+  form?: string;
+  label?: string;
+  chainId: CaipChainId;
+};
+
+export const SnapUIAddressInput: FunctionComponent<
+  SnapUIAddressInputProps & FormTextFieldProps<'div'>
+> = ({ name, form, label, chainId, error, ...props }) => {
+  const { handleInputChange, getValue, focusedInput, setCurrentFocusedInput } =
+    useSnapInterfaceContext();
+
+  const inputRef = useRef<HTMLDivElement>(null);
+
+  const initialValue = getValue(name, form) as string;
+
+  const [value, setValue] = useState(
+    initialValue
+      ? parseCaipAccountId(initialValue as CaipAccountId).address
+      : '',
+  );
+  const [matchedAddressName, setMatchedAddressName] = useState<string | null>(
+    null,
+  );
+
+  const accountsInfo: Record<string, string> = useSelector((state) =>
+    getAccountInfoByCaipChainId(state, chainId),
+  );
+
+  /*
+   * Focus input if the last focused input was this input
+   * This avoids losing the focus when the UI is re-rendered
+   */
+  useEffect(() => {
+    if (inputRef.current && name === focusedInput) {
+      (inputRef.current.querySelector('input') as HTMLInputElement).focus();
+    }
+  }, [inputRef]);
+
+  const getMatchedAddressName = (address: string) => {
+    const normalizedAddress = address.toLowerCase();
+    return accountsInfo[normalizedAddress];
+  };
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value);
+    const matchedName = getMatchedAddressName(event.target.value);
+    if (matchedName) {
+      setMatchedAddressName(matchedName);
+    }
+
+    const newValue = event.target.value
+      ? `${chainId}:${event.target.value}`
+      : null;
+
+    handleInputChange(name, newValue, form);
+  };
+
+  const handleFocus = () => setCurrentFocusedInput(name);
+  const handleBlur = () => setCurrentFocusedInput(null);
+
+  const handleClear = () => {
+    setValue('');
+    setMatchedAddressName(null);
+  };
+
+  const MatchedAccountInfo = () => {
+    return (
+      <Box display={Display.Flex} flexDirection={FlexDirection.Column}>
+        {label && (
+          <Label className={classnames('mm-form-text-field__label')}>
+            {label}
+          </Label>
+        )}
+        <Box
+          display={Display.Flex}
+          backgroundColor={BackgroundColor.backgroundDefault}
+          alignItems={AlignItems.center}
+          borderWidth={1}
+          borderRadius={BorderRadius.LG}
+          borderColor={BorderColor.borderMuted}
+          paddingLeft={4}
+          paddingRight={4}
+          gap={2}
+          style={{
+            height: '48px',
+          }}
+        >
+          <Jazzicon address={value} diameter={24} style={{ height: '24px' }} />
```

Comment:
This component is not compatible with non-EVM addresses, also we should allow use of both blockies and jazzicons. We should use `SnapUIAvatar`.